### PR TITLE
Improve validation on `ExpressionEvaluator::bind` and `Val::Val`, remove `executor_utils::validateKernelInputs`.

### PR DESCRIPTION
**Pull request by zasdfgbnm**
This PR adds validation on the concrete value when used to bind an `ExpressionEvaluator` or construct a `Val`. Also, I removed `executor_utils::validateKernelInputs` because it is redundant with the added validations and contains some dirty code needed to be cleaned up.

### CODE REVIEW

**Comment by zasdfgbnm** [comment]
!build

**Comment by zasdfgbnm** [comment]
!build

**Comment by zasdfgbnm** [comment]
!build

File: csrc/type.h [hunk 1]
```diff
@@ -305,6 +305,11 @@ DataType getComplexTypeFromType(DataType dtype);
 // Return if the datatype is supported on the current device
 TORCH_CUDA_CU_API bool isSupportedTypeByDevice(DataType dtype);
 
+TORCH_CUDA_CU_API int64_t dataTypeSize(DataType type);
```
Comment (zasdfgbnm) [inline csrc/type.h:308]: Just moved up.

**Review by naoyam — COMMENTED** [review]
Where do we do the equivalent checks as `validateKernelInputs`?

**Comment by zasdfgbnm** [comment]
> Where do we do the equivalent checks as `validateKernelInputs`?

`FusionExecutor::runFusion` binds all inputs. And when binding, the `validateValWithConcreteValue` does the check we need.

**Review by naoyam — APPROVED** [review]
LGTM. Thanks for the cleanup.

**Pull request merged**